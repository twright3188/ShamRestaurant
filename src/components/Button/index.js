import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import Colors from '../../theme';

const styles = StyleSheet.create({
  buttonBody: {
    minWidth: '100%',
    minHeight: 36,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 8,
    borderWidth: 1,
    borderColor: Colors.coloredButtonBackgroundColor,
  },
  buttonText: {
    color: Colors.buttonText,
    fontFamily: Colors.FontBaseMedium,
    fontWeight: '500',
    textAlign: 'center'
  }
});

function Button(props) {
  const { children, style, color, onPress, disabled, fontSize, ...otherProps } = props;

  return (
    <TouchableOpacity style={[styles.buttonBody, style, { opacity: disabled ? 0.2 : 1 }]} onPress={onPress} {...otherProps}>
      <Text
        style={[styles.buttonText, { color, fontSize }]} 
      >
        {children}
      </Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  style: PropTypes.shape({ root: PropTypes.string }),
  color: PropTypes.string,
  onPress: PropTypes.func,
  disabled: PropTypes.bool,
  fontSize: PropTypes.number,
};

Button.defaultProps = {
  children: '',
  style: {},
  color: Colors.buttonText,
  onPress: null,
  disabled: false,
  fontSize: 16
};

export default Button;