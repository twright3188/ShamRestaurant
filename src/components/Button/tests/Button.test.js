import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

it('button renders correctly', () => {
    const tree = renderer.create(<Button>View bouquets</Button>).toJSON();
    expect(tree).toMatchSnapshot();
});