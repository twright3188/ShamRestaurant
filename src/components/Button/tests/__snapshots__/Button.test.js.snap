// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#399AE5",
      "borderRadius": 8,
      "borderWidth": 1,
      "justifyContent": "center",
      "minHeight": 36,
      "minWidth": "100%",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#399AE5",
          "fontFamily": "Helvetica Neue",
          "fontWeight": "500",
          "textAlign": "center",
        },
        Object {
          "color": "#399AE5",
          "fontSize": 16,
        },
      ]
    }
  >
    View bouquets
  </Text>
</View>
`;
